android/app/src/main/java/com/mattermost/rnbeta/NotificationsLifecycleFacade.java

```diff
@@ -10,18 +16,67 @@
 import com.wix.reactnativenotifications.core.AppLifecycleFacade;
 
 import java.util.Set;
+import 	android.util.ArraySet;
 import java.util.concurrent.CopyOnWriteArraySet;
+import android.view.WindowManager.LayoutParams;
+
 
 public class NotificationsLifecycleFacade extends ActivityCallbacks implements AppLifecycleFacade {
 
     private static final String TAG = NotificationsLifecycleFacade.class.getSimpleName();
+    private Bundle managedConfig = null;
 
     private Activity mVisibleActivity;
     private Set<AppVisibilityListener> mListeners = new CopyOnWriteArraySet<>();
 
+    private final IntentFilter restrictionsFilter =
+            new IntentFilter(Intent.ACTION_APPLICATION_RESTRICTIONS_CHANGED);
+
+    private final BroadcastReceiver restrictionsReceiver = new BroadcastReceiver() {
+        @Override public void onReceive(Context context, Intent intent) {
+
+            if (mVisibleActivity != null) {
+                // Get the current configuration bundle
+                RestrictionsManager myRestrictionsMgr =
+                    (RestrictionsManager) mVisibleActivity
+                            .getSystemService(Context.RESTRICTIONS_SERVICE);
+                managedConfig = myRestrictionsMgr.getApplicationRestrictions();
+
+                // Check current configuration settings, change your app's UI and
+                // functionality as necessary.
+                Log.i("ReactNative", "Managed Configuration Changed");
+                MattermostManagedModule.instance.sendConfigChanged(managedConfig);
+            }
+        }
+    };
+
+    @Override
+    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+        if (MattermostManagedModule.instance.isBlurAppScreenEnabled()) {
+            activity.getWindow().setFlags(LayoutParams.FLAG_SECURE,
+                    LayoutParams.FLAG_SECURE);
+        }
+        if (managedConfig!= null && managedConfig.size() > 0) {
+            activity.registerReceiver(restrictionsReceiver, restrictionsFilter);
+        }
+    }
+
     @Override
     public void onActivityResumed(Activity activity) {
         switchToVisible(activity);
+
+        if (managedConfig != null && managedConfig.size() > 0) {
+            RestrictionsManager myRestrictionsMgr =
+                    (RestrictionsManager) activity
+                            .getSystemService(Context.RESTRICTIONS_SERVICE);
+
+            Bundle newConfig = myRestrictionsMgr.getApplicationRestrictions();
+            if (!equalBundles(newConfig ,managedConfig)) {
+                Log.i("ReactNative", "onResumed Managed Configuration Changed");
+                managedConfig = newConfig;
+                MattermostManagedModule.instance.sendConfigChanged(managedConfig);
```

Comment:
Would it make more sense for `sendConfigChanged(...)` to be a method in this class, rather than in `MattermostManagedModule`? Unless it's being called from multiple places, it would be cleaner to avoid the static access and just do it here.